src/compiler/chainrules.jl

```diff
@@ -10,13 +17,20 @@ If it does not, then the second argument is a list of edges to attach to the Cod
 such that if a suitable rule is defined later, the generated function will recompile.
 """
 function has_chain_rrule(T)
-  m = meta(Tuple{typeof(rrule),T.parameters...})
-  if m.method !== chainrules_fallback
-    # found a rrule, no need to add any edges
-    return true, nothing
+  config_T, arg_Ts = Iterators.peel(T.parameters)
+  m_with_config = meta(Tuple{typeof(rrule), config_T, arg_Ts...})
+  if m_with_config.method === rrule_redispatcher_method
+    # it is being redispatched without config, so check it that hits the fallback
+    m_without_config = meta(Tuple{typeof(rrule), arg_Ts...})
+    if m_without_config.method === rrule_fallback_method
+      # no rrule exists, return instance for m_with_config as that will be invalidated 
+      # directly if configured rule added, or indirectly if unconfigured ruleadded
```

Comment:
I don't know what this comment means

Reply:
```suggestion
      # directly if configured rule added, or indirectly if unconfigured rule added
```

> I don't know what this comment means

It is about how julia knows when to trigger recompilation because of new methods being added.
We can  have a call about it, it needs a fair bit of background info